Servers/SystemServer/Service.cpp

```diff
@@ -239,6 +239,13 @@ void Service::spawn()
             }
         }
 
+        if (!m_working_directory.is_empty()) {
```

Comment:
* Other strings here are checked for `is_null()` rather than `is_empty()`, let's do the same for consistency. It's unlikely that an empty path to the working directory could be meaningful, but let's allow services to specify that and attempt to pass it to the kernel; then the kernel would do something about it (like reject it).
* Let's `chdir()` before dropping privileges.